lib/rubygems/ext/cargo_builder.rb

```diff
@@ -127,42 +145,19 @@ def rustc_dynamic_linker_flags(dest_dir)
     split_flags("DLDFLAGS")
       .map {|arg| maybe_resolve_ldflag_variable(arg, dest_dir) }
       .compact
-      .flat_map {|arg| ldflag_to_link_modifier(arg, dest_dir) }
+      .flat_map {|arg| ldflag_to_link_modifier(arg) }
   end
 
   def rustc_lib_flags(dest_dir)
-    split_flags("LIBS").flat_map {|arg| ldflag_to_link_modifier(arg, dest_dir) }
+    split_flags("LIBS").flat_map {|arg| ldflag_to_link_modifier(arg) }
   end
 
   def split_flags(var)
     Shellwords.split(RbConfig::CONFIG.fetch(var, ""))
   end
 
-  def ldflag_to_link_modifier(arg, dest_dir)
-    flag = arg[0..1]
-    val = arg[2..-1]
-
-    case flag
-    when "-L" then ["-L", "native=#{val}"]
-    when "-l" then ["-l", val.to_s]
-    when "-F" then ["-l", "framework=#{val}"]
-    else ["-C", "link_arg=#{arg}"]
-    end
-  end
-
-  def link_flag(link_name)
-    # These are provided by the CRT with MSVC
-    # @see https://github.com/rust-lang/pkg-config-rs/blob/49a4ac189aafa365167c72e8e503565a7c2697c2/src/lib.rs#L622
-    return [] if msvc_target? && ["m", "c", "pthread"].include?(link_name)
-
-    if link_name.include?("ruby")
-      # Specify the lib kind and give it the name "ruby" for linking
-      kind = ruby_static? ? "static" : "dylib"
-
-      ["-l", "#{kind}=ruby:#{link_name}"]
-    else
-      ["-l", link_name]
-    end
+  def ldflag_to_link_modifier(arg)
+    LinkFlagConverter.call(arg)
```

Comment:
I think there is no need to create this additional class just to hide one method. Also I'm not big fan of generic `call` methods. IMHO the purpose of `call` method naming convention was to be able to mimic `proc` and its `call` method, which is not used in here. Is there any problem to revive the method back to this class?

Reply:
I want to be able to isolate it for more robust testing, so having it in a separate class is nice for that. Is there another method name besides `call` you would prefer??

Reply:
I renamed the method to `convert_to_cargo_flag`, does that work for you??

Reply:
looks ok :heart:

Reply:
> I want to be able to isolate it for more robust testing, so having it in a separate class is nice for that. Is there another method name besides `call` you would prefer??

Since the purpose of the class was to convert something, I would (personally) prefer to use `convert` method for example.